addons/mail/tools/discuss.py

```diff
@@ -37,29 +37,39 @@ def wrapper(self, *args, **kwargs):
 
     return wrapper
 
-def get_twilio_credentials(env) -> (str, str):
+
+def get_twilio_credentials(env) -> tuple[str, str] | None:
     """
     To be overridable if we need to obtain credentials from another source.
     :return: tuple(account_sid: str, auth_token: str)
     """
     params = env["ir.config_parameter"].sudo()
     account_sid = params.get_param("mail.twilio_account_sid")
     auth_token = params.get_param("mail.twilio_account_token")
+    if not params.get_param("mail.use_twilio_rtc_servers"):
+        return None
     return account_sid, auth_token
 
 
 def get_sfu_url(env) -> str | None:
-    sfu_url = env['ir.config_parameter'].sudo().get_param("mail.sfu_server_url")
+    params = env["ir.config_parameter"].sudo()
+    sfu_url = params.get_param("mail.sfu_server_url")
     if not sfu_url:
         sfu_url = os.getenv("ODOO_SFU_URL")
+    if not params.get_param("mail.use_sfu_server"):
+        return None
     if sfu_url:
         return sfu_url.rstrip("/")
+    return None
 
 
 def get_sfu_key(env) -> str | None:
-    sfu_key = env['ir.config_parameter'].sudo().get_param('mail.sfu_server_key')
+    params = env["ir.config_parameter"].sudo()
+    sfu_key = params.get_param('mail.sfu_server_key')
     if not sfu_key:
-        return os.getenv("ODOO_SFU_KEY")
+        sfu_key = os.getenv("ODOO_SFU_KEY")
+    if not params.get_param("mail.use_sfu_server"):
```

Comment:
no need to modify this function, we won't try to connect to the sfu if `get_sfu_url` does not return an URL anyways